packages/core/src/blueprint/plugins/press.tmpl.js

```diff
@@ -56,15 +46,14 @@ export default async (ctx, inject) => {
       }
     }
   } else {
-    press.get = function get (url) {
-      return ctx.$http.$get(url).catch(err => consola.warn(err))
+    plugin.get = function get (url) {
+      return context.$http.$get(url).catch(err => consola.warn(err))
     }
   }
 
-  if (!ctx.$press) {
-    ctx.$press = press
-    inject('press', press)
-  }
-
-  await pressMiddleware(ctx, true)
-}
+  // this is a workaround to prevent hydration errors
+  // due to middlewares not running on first load on the client
+  // TODO: try to understand why returning the pressMiddleware
+  // promise doesnt work and this has to be async/await
```

Comment:
@pimlie probably because the function is already async?

Reply:
Which function? Im thinking it could be a transpile issue, but havent searched for the transpiled code yet

Eg this should still work, use that often. Mainly because its an eslint issue to return & await at the sametime
```js
async function test() {
  await something()
  return new Promise()
  // next line triggers an eslint warning
  //return await something()
}